shared/common-adapters/plain-input.stories.js

```diff
@@ -17,6 +19,138 @@ const commonProps = {
   style: {borderWidth: 1, borderStyle: 'solid', borderColor: globalColors.black_10},
 }
 
+type TestInputState = {
+  value: string,
+}
+class TestInput extends React.Component<{}, TestInputState> {
+  state = {value: ''}
+  // prettier-ignore
+  _input = React.createRef<typeof PlainInput>()
+
+  _insertText = (t: string) => {
+    const input = this._input.current
+    if (input) {
+      const selection = input.getSelection()
+      if (selection) {
+        this.setState(
+          s => {
+            const value = s.value.substring(0, selection.start) + t + s.value.substring(selection.start)
+            return {value}
+          },
+          () => {
+            const input = this._input.current
+            if (input) {
+              const newCursorPos = selection.start + t.length
+              input.setSelection({start: newCursorPos, end: newCursorPos})
+              input.focus()
+            }
+          }
+        )
+      }
+    }
+  }
+
+  render() {
+    return (
+      <Box2 direction="vertical" fullWidth={true} gap="small">
+        <PlainInput
+          {...commonProps}
+          ref={this._input}
+          value={this.state.value}
+          onEnterKeyDown={() => this._insertText('foo')}
+          onChangeText={v => this.setState(s => (s.value === v ? null : {value: v}))}
+        />
+        <ButtonBar>
+          <Button type="Secondary" label="Insert 'foo' (Enter)" onClick={() => this._insertText('foo')} />
+        </ButtonBar>
+      </Box2>
+    )
+  }
+}
```

Comment:
mind also adding a multiline version of this playground? sadly, I had found that some bugs only show up with multiline inputs...